src/linux/Payload.Linux/build.sh

```diff
@@ -32,17 +38,106 @@ case "$i" in
 CONFIGURATION="${CONFIGURATION:=Debug}"
 if [ -z "$VERSION" ]; then
     die "--version was not set"
 fi
 
-PAYLOAD="$PAYLOAD_OUT/tar/$CONFIGURATION/payload"
-TAROUT="$PAYLOAD_OUT/tar/$CONFIGURATION/gcmcore-linux-x86_64-$VERSION.tar.gz"
+ARCH="`dpkg-architecture -q DEB_HOST_ARCH`"
+if test -z "$ARCH"; then
+  die "Could not determine host architecture!"
+fi
+
+PAYLOAD="$PAYLOAD_OUT/payload/$CONFIGURATION"
+TAROUT="$PAYLOAD_OUT/gcmcore-linux_$ARCH.$CONFIGURATION.$VERSION.tar.gz"
+DEBPKG="$PAYLOAD_OUT/gcmcore-linux/"
+DEBOUT="$PAYLOAD_OUT/gcmcore-linux_$ARCH.$CONFIGURATION.$VERSION.deb"
+SYMBOLOUT="$PAYLOAD.sym"
+
+# Cleanup any old payload directory
+if [ -d "$PAYLOAD" ]; then
+    echo "Cleaning old payload directory '$PAYLOAD'..."
+    rm -rf "$PAYLOAD"
+fi
+
+# Ensure directories exists
+mkdir -p "$PAYLOAD" "$SYMBOLOUT" "$DEBPKG"
+
+# Publish core application executables
+echo "Publishing core application..."
+dotnet publish "$GCM_SRC" \
+	--configuration="$CONFIGURATION" \
+	--framework="$FRAMEWORK" \
+	--runtime="$RUNTIME" \
+    --self-contained=true \
+    "/p:PublishSingleFile=True" \
+	--output="$(make_absolute "$PAYLOAD")" || exit 1
+
+# Collect symbols
+echo "Collecting managed symbols..."
+mv "$PAYLOAD"/*.pdb "$SYMBOLOUT" || exit 1
+
+echo "Build complete."
+
+#####################################################################
+# PACKING
+#####################################################################
+echo "Packing Payload.Linux..."
+# Cleanup any old archive files
+if [ -e "$TAROUT" ]; then
+    echo "Deleteing old archive '$TAROUT'..."
+    rm "$TAROUT"
+fi
+
+# Ensure the parent directory for the archive exists
+mkdir -p "$(dirname "$TAROUT")"
+
+# Set full read, write, execute permissions for owner and just read and execute permissions for group and other
+echo "Setting file permissions..."
+/bin/chmod -R 755 "$PAYLOAD" || exit 1
+
+# Build tarball
+echo "Building archive..."
+pushd "$PAYLOAD"
+tar -czvf "$TAROUT" * || exit 1
+popd
+
+# Build .deb
+INSTALL_TO="$DEBPKG/usr/bin/"
+mkdir -p "$DEBPKG/DEBIAN" "$INSTALL_TO"
+
+# make the debian control file
+cat >"$DEBPKG/DEBIAN/control" <<EOF
+Package: gcmcore
+Version: $VERSION
+Section: vcs
+Priority: optional
+Architecture: $ARCH
+Depends:
+Maintainer: GCM-Core <gcmcore@microsoft.com>
+Description: Cross Platform Git-Credential-Manager-Core command line utility.
+ Linux build of the GCM-Core project to support auth with a number of
+ git hosting providers including GitHub, BitBucket, and Azure DevOps.
+ Hosted at https://github.com/microsoft/Git-Credential-Manager-Core
+EOF
+
+# Copy single binary to target installation location
+cp "$PAYLOAD/git-credential-manager-core" "$INSTALL_TO"
 
-# Layout and pack
-"$PAYLOAD_SRC/layout.sh" --configuration="$CONFIGURATION" --output="$PAYLOAD" || exit 1
-"$PAYLOAD_SRC/pack.sh" --payload="$PAYLOAD" --output="$TAROUT" || exit 1
+dpkg-deb --build "$DEBPKG" "$DEBOUT"
 
-echo "Build of Payload.Linux complete."
+echo "Pack complete."
```

Comment:
Is it possible to publish the `.deb` as a build artifact? I'd love to download the `.deb` from a PR and install it locally for testing.

Reply:
Yep!